templates/common/.vscode/setup-api.sh

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+
+function codespacesPortUrl() {
+    local portNumber=$1
+    gh codespace ports \
+        -c "$CODESPACE_NAME" \
+        --json sourcePort,browseUrl \
+        --jq "map(select(.sourcePort == $portNumber))[0].browseUrl"
+}
+
+function getEnvFile() { 
+    azd env list --output json | jq -r "map(select(.IsDefault == true))[].DotEnvPath"
+}
+
+envFile="$(getEnvFile)"
+backupEnvFile="$envFile.backup"
+debugEnvFile="$envFile.debug"
+
+if [ -f "$envFile" ]; then 
+    echo "Copying .env file to backup" 
+    cp "$envFile" "$backupEnvFile"
+fi
+
+if [ -f "$debugEnvFile" ]; then
+    echo "Existing debug envfile, using that"
+    mv "$debugEnvFile" "$envFile"
+fi
+
+if [ "$CODESPACES" = 'true' ]; then
+    echo "Running in Codespaces. Setting port configurations."
+
+    apiPortUrl=$(codespacesPortUrl 3100)
+    echo "azd env set REACT_APP_API_BASE_URL \"$apiPortUrl\""
+    azd env set REACT_APP_API_BASE_URL "$apiPortUrl"
```

Comment:
Thanks! We do want to set `REACT_APP_API_BASE_URL` when launching debug of the API (otherwise we're connecting to the deployed endpoint). 

`REACT_APP_WEB_BASE_URL` logic is indeed redundant and I'll take it out. 

Incidentally this gives us the ability to "clean up" the `.env.debug` file after API debugging ends. The frontend will connect to the deployed endpoint _unless_ there is an active API debugging session.